UnityProject/Assets/Scripts/UI/Systems/AdminTools/DevTools/DeviceMover.cs

```diff
@@ -0,0 +1,164 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Logs;
+using Shared.Managers;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+[RequireComponent(typeof(EscapeKeyTarget))]
+public class DeviceMover : SingletonManager<DeviceMover>
+{
+	//TODO on  appear at World position  reset stuff
+
+
+    public Button StopSelectingButton;
+
+    public Toggle RoundToggle;
+
+	// so we can escape while drawing - enabled while drawing, disabled when done
+	private EscapeKeyTarget escapeKeyTarget;
+
+	public GameObject PressedObject = null;
+
+	public GameGizmoLine CursorLine;
+
+	public bool Updating = false;
+
+	public Vector3 StartPositionWorld;
+
+	private void OnEnable()
+	{
+		escapeKeyTarget = GetComponent<EscapeKeyTarget>();
+	}
+
+	public override void Start()
+	{
+		base.Start();
+		this.gameObject.SetActive(false);
+	}
+
+	private void OnDisable()
+	{
+		OnEscape();
+		if (Updating)
+		{
+			UpdateManager.Remove(CallbackType.UPDATE, UpdateMe);
+			Updating = false;
+		}
+	}
+
+
+	private void UpdateMe()
```

Comment:
the contents of this function can be split into smaller ones. 

https://plugins.jetbrains.com/plugin/12024-cognitivecomplexity

this should help